File: iOS-App/CleaningShiftManager/Models/Notification.swift
//
//  Notification.swift
//  CleaningShiftManager
//

import Foundation

struct AppNotification: Codable, Identifiable {
    let id: String
    let type: String
    let title: String
    let message: String
    let read: Bool
    let createdAt: String
    let relatedShift: String?
    let shiftDetails: ShiftDetails?
    let timeChangeDetails: TimeChangeDetails?
    
    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case type
        case title
        case message
        case read
        case createdAt
        case relatedShift
        case shiftDetails
        case timeChangeDetails
    }
    
    struct ShiftDetails: Codable {
        let apartmentName: String
        let apartmentAddress: String
        let scheduledDate: String
        let scheduledStartTime: String
        let scheduledEndTime: String?
        let confirmed: Bool
    }
    
    struct TimeChangeDetails: Codable {
        let newStartTime: String
        let newEndTime: String?
        let reason: String?
        let operatorConfirmed: Bool
    }
}

struct NotificationsResponse: Codable {
    let notifications: [AppNotification]
}

File: Android-App/app/src/main/java/com/cleaningmanager/app/data/models/Notification.kt
package com.cleaningmanager.app.data.models

import com.google.gson.annotations.SerializedName

data class AppNotification(
    @SerializedName("_id")
    val id: String,
    val type: String,
    val title: String,
    val message: String,
    val read: Boolean,
    val createdAt: String,
    val relatedShift: String? = null,
    val shiftDetails: ShiftDetails? = null,
    val timeChangeDetails: TimeChangeDetails? = null
)

data class ShiftDetails(
    val apartmentName: String,
    val apartmentAddress: String,
    val scheduledDate: String,
    val scheduledStartTime: String,
    val scheduledEndTime: String? = null,
    val confirmed: Boolean
)

data class TimeChangeDetails(
    val newStartTime: String,
    val newEndTime: String? = null,
    val reason: String? = null,
    val operatorConfirmed: Boolean
)

data class NotificationsResponse(
    val notifications: List<AppNotification>
)

data class MarkReadRequest(
    val notificationIds: List<String>,
    val read: Boolean
)


